### PR TITLE
Added the fts star to the scoreboard

### PR DESCRIPTION
**Pull request by Mond1c**

### CODE REVIEW

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 1]
```diff
@@ -115,15 +117,36 @@ const TaskResultLabelWrapper2 = styled.div`
   color: #fff;
 `;
 
-// TODO: fts start
-const ICPCTaskResultLabel2 = ({ problemResult: r, ...props }) => {
+
+const StarIconWrap = styled.div`
+    position: absolute;
+    width: 33px;
+    height: 33px;
```
Comment (Mond1c) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:124]: Do I need to make this value constant? (maybe STAR_SIZE in config.js?)

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 2]
```diff
@@ -115,15 +117,36 @@ const TaskResultLabelWrapper2 = styled.div`
...
+
+const AttemptsLabelWrapper = styled.div`
+    position: relative;
+`;
+
+const ICPCTaskResultLabel2 = ({ problemColor, problemResult: r, ...props }) => {
     const status = getStatus(r.isFirstToSolve, r.isSolved, r.pendingAttempts, r.wrongAttempts);
     const attempts = r.wrongAttempts + r.pendingAttempts;
+    const defaultColorForStar = "#000";
+    const greenColor = "#3bba6b";
```
Comment (Mond1c) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:142]: Do we need to check for green color. If so, I will not delete it.

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 1 above)
Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:124]: Yes, please extract to config.js

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 3]
```diff
@@ -115,27 +117,52 @@ const TaskResultLabelWrapper2 = styled.div`
...
+const StarIconWrap = styled.div`
+    position: absolute;
+    width: 33px;
+    height: 33px;
+    mask: url(${star});
+    background-position: center;
+    background-color: ${({ color }) => color};
+`;
+
+const StarIcon = ({ color }) => {
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:130]: There is no reason for StarIcon component. Just rename StarIconWrap to StarIcon

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 4]
```diff
@@ -115,27 +117,52 @@ const TaskResultLabelWrapper2 = styled.div`
...
+    return <StarIconWrap color={color}/>;
+};
+
+const AttemptsOrScoreLabelWrapper = styled.div`
+    position: relative;
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:135]: Why is there a rouge position: relative?

Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:135]: Doesn't it  work without the position: relative.

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 5]
```diff
@@ -115,27 +117,52 @@ const TaskResultLabelWrapper2 = styled.div`
...
+        </AttemptsOrScoreLabelWrapper>
     </TaskResultLabelWrapper2>;
 };
-export const TaskResultLabel = memo(({ problemResult, minScore, maxScore, ...props }) => {
+export const TaskResultLabel = memo(({ problemColor, problemResult, minScore, maxScore, ...props }) => {
     return <>
-        {problemResult.type === "ICPC" && <ICPCTaskResultLabel2 problemResult={problemResult} {...props}/>}
-        {problemResult.type === "IOI" && <IOITaskResultLabel2 problemResult={problemResult} minScore={minScore} maxScore={maxScore} {...props}/>}
+        {problemResult.type === "ICPC" && <ICPCTaskResultLabel2 problemColor={problemColor} problemResult={problemResult} {...props}/>}
+        {problemResult.type === "IOI" && <IOITaskResultLabel2 problemColor={problemColor} problemResult={problemResult} minScore={minScore} maxScore={maxScore} {...props}/>}
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:166]: No reason to extract problemColor or problemResult or minScore or maxScore from the {...props}.
Just leave {...props}

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx [hunk 6]
```diff
@@ -115,27 +117,52 @@ const TaskResultLabelWrapper2 = styled.div`
...
+            <AttemptsOrScoreLabelWrapper>
+                {TeamTaskSymbol[status]}
+                {status !== TeamTaskStatus.untouched && attempts > 0 && attempts}
+            </AttemptsOrScoreLabelWrapper>
         </TaskResultLabelWrapper2>
     </>;
 };
 
-const IOITaskResultLabel2 = ({ problemResult: r, minScore, maxScore,  ...props }) => {
+const IOITaskResultLabel2 = ({ problemColor, problemResult: r, minScore, maxScore,  ...props }) => {
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/atoms/ContestLabels.jsx:155]: Is it really necessary to rename problemResult to r?

File: src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx [hunk 7]
```diff
@@ -119,7 +119,7 @@ export const ScoreboardRow = ({ teamId, hideTasks, optimismLevel }) => {
             formatPenalty(scoreboardData?.penalty)
         } />}
         {!hideTasks && scoreboardData?.problemResults.map((result, i) =>
-            <ScoreboardTaskResultLabel problemResult={result} key={i}
+            <ScoreboardTaskResultLabel problemResult={result} key={i} problemColor={contestData.problems[i].color}
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx:122]: Maybe we should create a contest-aware component that will look up the colors from the store, but all we had to do is pass it the problem number?
Something like
```javascript
export const ProblemLabel = ({problemIndex, className}) => {
    const problemData = useSelector((state) => state ...... );
    return <TaskResultLabel problemColor={problemData.color} ..... >
}
```

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 6 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:155]: I didn't to it. That was before I started pr.

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 4 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:135]: If I remove it, the star will above the text

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 3 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:130]: fixed

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 4 above)
Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:135]: Change the order of the elements in the html tree

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 6 above)
Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:155]: Then I guess we will refactor it some time in the future.

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 5 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:166]: I can't extract problemResult because:
```js
problemResult.type === "ICPC"
```

Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:166]: minScore, maxScore, problemColor are now in props

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 1 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx:124]: fixed

File: src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx (hunk 7 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx]: But we are already getting minScore, maxScore, etc. here. Do we want to create a new component for this?

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 4 above)
Comment (Mond1c) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx]: I just changed ```css position: relative``` to ```css position: absolute``` and removed ```css position: absolute``` from the StarIcon div.

File: src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx (hunk 7 above)
Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx]: Yes. Here we are also getting a lot of other useless data. We should be more specific to make things more pure

File: src/frontend/overlay/src/components/atoms/ContestLabels.jsx (hunk 4 above)
Comment (irdkwmnsb) [reply src/frontend/overlay/src/components/atoms/ContestLabels.jsx]: Ok, then just leave it

File: src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx [hunk 8]
```diff
@@ -119,7 +119,7 @@ export const ScoreboardRow = ({ teamId, hideTasks, optimismLevel }) => {
             formatPenalty(scoreboardData?.penalty)
         } />}
         {!hideTasks && scoreboardData?.problemResults.map((result, i) =>
-            <ScoreboardTaskResultLabel problemResult={result} key={i} problemColor={contestData.problems[i].color}
+            <ScoreboardTaskResultLabel problemResult={result} key={i} problemColor={contestData?.problems[i]?.color}
```
Comment (irdkwmnsb) [inline src/frontend/overlay/src/components/organisms/widgets/Scoreboard.jsx:122]: This is the thing I don't like. Passing too many props from the state in the context that is not suitable. I could understand if this was a specialized component that is extracting data from the store and displaying the data, but when it's used here - I'm not sure how I feel about that.

**Review by irdkwmnsb — APPROVED** [review]

**Pull request merged**